src/system/SystemError.cpp

```diff
@@ -229,20 +156,12 @@ DLL_EXPORT CHIP_ERROR MapErrorLwIP(err_t aError)
  */
 DLL_EXPORT const char * DescribeErrorLwIP(CHIP_ERROR aError)
 {
-    if (!IsErrorLwIP(aError))
+    if (!ChipError::IsRange(ChipError::Range::kLwIP, aError))
     {
         return nullptr;
     }
 
-    // Error might be a signed or unsigned type.  But we know the value is no
-    // larger than CHIP_SYSTEM_LWIP_ERROR_MAX and that this means it's safe to
-    // store in int.
-    static_assert(INT_MAX > CHIP_SYSTEM_LWIP_ERROR_MAX, "Our subtraction will fail");
-    const int lErrorWithoutOffset = aError - CHIP_SYSTEM_LWIP_ERROR_MIN;
-    // Cast is safe because the range from CHIP_SYSTEM_LWIP_ERROR_MIN to
-    // CHIP_SYSTEM_LWIP_ERROR_MAX all fits inside err_t.  See static_assert in
-    // MapErrorLwIP.
-    const err_t lError = static_cast<err_t>(-lErrorWithoutOffset);
+    const err_t lError = static_cast<err_t>(-ChipError::GetValue(aError));
```

Comment:
```suggestion
    const err_t lError = -static_cast<err_t>(ChipError::GetValue(aError));
```
to undo the operations we did in MapErrorLwIP, where we negated, then cast.

Reply:
Ack